docs/apache-airflow/core-concepts/overview.rst

```diff
@@ -43,13 +43,26 @@ An Airflow installation generally consists of the following components:
 Basic airflow architecture
 --------------------------
 
-This is the basic architecture of Airflow that you'll see in simple installations:
+This is the basic architecture of Airflow that you'll see in simple installations.
+
+Note that even in the simple installation, Airflow Webserver does not access the DAG files directly
+(the code you see in the Code tab of the UI is synchronized via the metadata database).
+
+This is to allow for a more secure deployment, where the webserver has never the need or even
+possibility to executed code submitted by DAG authors, the only custom code that webserver can
+execute are the plugins and those should be only installed and deployed by the Airflow Deployment Manager.
```

Comment:
Would leave this out.

Reply:
This is a source of major confusion. Many our users still think that webserver accesses the DAG files and needs it even 3 years  after releasing Airflow 2. I've been answering at least 10 times to our users that this is not the case as they still continued to deploy Airflow webserver WITH DAG folder even if this was useless.

I believe we failed to communicate that - that's why I added it here.

Reply:
Thought a bit and I think I will simply add much simpler diagram very much like the original one - showing "self-contained" installation - without workers and triggerers and with single user and make it the first diagram - with explanation that this is the simplest configuration and explaination that it only serves very basic installation. This way the diagrams will get gradually more complex  and showing "true" nature of some of the decisions made, while not confusing the users who **just** want to start their journey with Airflow.

I believe that will satisfy both - your desire @BasPH  to have things simple and my desire to use the architecture diagrams of ours to explain some design decisions we have and to explain those who are looking for self-deployment of Airflow and want to understand our intentions.

WDYT?

Reply:
I agree with @potiuk - but maybe this is currently also in the light of many security reviews. If no agreement, then the reasoning can be shortened a bit.

Reply:
OK. Let me know @BasPH and @jscheffl what you think now. We should - of course - tweak it still a bit (better icons as discussed below and possibly play a bit with graphviz/diagram layout to add some margins and get it a bit more readable. Plus I can add a few links to other parts of the documentation when I am mentioning other pieces - to get it all nicely hyperlinked. 

But I **think** this one might achieve good balance between the simplicity and being able to explain concepts and architecture in "gradually more complex" ways - without overloading the users who just want to install airflow and use it locally.

Reply:
I **think** this might be resolved - please reopen if you think it's not.

Reply:
@potiuk Sounds good. It's not my goal to keep things simple per se. However, I do think we should introduce complex topics more gradually, one concept at a time. A piece of text, or an image, becomes difficult to comprehend if multiple previously unmentioned concepts are introduced at the same time to the reader.

So starting with a bare minimum diagram, and then saying "here's how a more complex setup might look like" is indeed a good approach to introduce the reader to a complex topic.